myapp/app/models/task.rb

```diff
@@ -3,10 +3,23 @@
 class Task < ApplicationRecord
   SORT_COLUMN_ALLOWED = %w[created_at end_date].freeze
 
+  enum status: { not_started: 0, in_progress: 1, done: 2 }
+
   validates :title, presence: true, length: { maximum: 40 }
   validates :description, length: { maximum: 500 }
 
   scope :sort_tasks, lambda { |sort_column, sort_direction|
     order("#{sort_column} #{sort_direction}") if sort_column.in? SORT_COLUMN_ALLOWED
   }
+
+  def self.search(conditions)
+    # 検索
+    tasks = Task.where('title LIKE?', "%#{conditions[:keyword]}%")
+    tasks = tasks.where('status=?', conditions[:status]) if conditions[:status] =~ /^[0|1|2]$/
```

Comment:
`conditions[:status]` の存在確認は正規表現ではなく↓でも良さそうですね

```ruby
conditions[:status].present?
```

Reply:
ありがとうございます。
そもそもstatusを０〜２の整数に制限して、それ以外は検索条件を無効にさせたいので
```suggestion
    tasks = tasks.where('status=?', conditions[:status]) if conditions[:status].in? Task.statuses.values.map!{|v| v.to_s}
```
にします。